### ISSUE TITLE
View-once photo may be opened for the second time

### ISSUE THREAD

**Issue by prawilny** [signalapp/Signal-Android#12229]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
View once photo opened on Android may be opened once again on desktop.

### Steps to reproduce
(Possible to reproduce both in 1:1 conversations and in groups)
- receive a view-once photo
- open this photo on Android
- open this photo on Desktop

**Actual result:** photo is opened.
**Expected result:** the media should be marked as "viewed" and impossible to open again. It works this way when opened on desktop first. I think it also used to work this way when opening on android first.

#### Android
**Device:** OnePlus Nord 2
**Android version:** 11
**Signal version:** 5.36.3

#### Desktop
**Signal version:** 5.41.0 (Flatpak from Flathub)

### Link to debug log
Android: https://debuglogs.org/android/5.36.3/306c7bf0813ffdaf6caf32623f449db33cb70481e993e8b1bd6b506a67d4a1f2
Linux: https://debuglogs.org/desktop/5.41.0/656381cd761e917e34171b10f7a3682bb3ecfc27b49ec03d3a72329f8679243f.gz

**Comment by johanw666** [comment]
I assume the message from Android that the image is opened isn't received by desktop. If you disconnect the phone from internet before opening you should get the same result.

**Comment by prawilny** [comment]
No, that's not the case. I had both the clients open simultaneously.

**Comment by cody-signal** [comment]
Thanks, looking into it.
